### PR TITLE
Move RLLib GPU release test dependencies to ml docker

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

cc @AmeerHajAli @sven1977 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by krfricke** [comment]
This change currently fails with numpy.

```
  - pip uninstall -y numpy ray || true
  - sudo rm -rf /home/ray/anaconda3/lib/python3.7/site-packages/numpy
  - pip3 install numpy==1.19.5 || true
```

seems to be still necessary. We should remove this in a future PR?

**Review by sven1977 — APPROVED** [review]
Looks great! Thanks @krfricke :)

**Comment by sven1977** [comment]
Actually, lots of tests are failing now due to the Atari ROMs missing. Could you take another look?

**Comment by AmeerHajAli** [comment]
can we also please remove the post build commands?

File: release/rllib_tests/app_config.yaml [hunk 1]
```diff
@@ -13,7 +12,3 @@ post_build_cmds:
   - sudo rm -rf /home/ray/anaconda3/lib/python3.7/site-packages/numpy
   - pip3 install numpy==1.19.5 || true
```
Comment (AmeerHajAli) [inline release/rllib_tests/app_config.yaml:13]: Can we figure out a way to remove line 12 and 13 please?

Comment (krfricke) [reply release/rllib_tests/app_config.yaml:13]: Yes, I was going to raise this in the product repo. This does not come up in the Docker container at runtime, only during build of cluster envs, so this is likely an issue with caching.

Comment (AmeerHajAli) [reply release/rllib_tests/app_config.yaml:13]: why can't we bake this into the docker files?

Comment (AmeerHajAli) [reply release/rllib_tests/app_config.yaml:13]: This is one of the pain points for our existing users. It won't work unless we fix this to not require post build commands.
If you think about how a customer sees this, then they will think that ray-ml is all they need.

Comment (krfricke) [reply release/rllib_tests/app_config.yaml:13]: It makes sense. I'll merge this PR first and will follow up with the product team. I don't think this has to do with the `rayproject/ray-ml` docker image, but with the `anyscale/` one

**Pull request merged**